### ISSUE TITLE
[BUG] - Autocomplete don't focus on input

### ISSUE THREAD

**Issue by iPagar** [nextui-org/nextui#3203]
### NextUI Version

2.4.1

### Describe the bug

When I click on autocomplete it should focus on input

### Your Example Website or App

https://codesandbox.io/p/sandbox/objective-nash-lff3pt

### Steps to Reproduce the Bug or Issue

1. Copy code from https://nextui.org/docs/components/autocomplete#usage
2. Click on autocomplete
3. Input is not focused 

### Expected behavior

Input has to be focused as it shown at https://nextui.org/docs/components/autocomplete#usage

### Operating System Version

macOS

### Browser

Chrome

**Comment by wingkwong** [comment]
seems it looks fine in local storybook but not in app.

**Comment by MarcoRodasW** [comment]
Same issue for me, i have autocompletes everywhere in the app 🥲

**Comment by MarcoRodasW** [comment]
@wingkwong Any temporary solution until it is fixed?

**Comment by daniel-esserweb** [comment]
This still not fixed in v2.4.2

**Comment by MarcoRodasW** [comment]
@daniel-esserweb Noup, I switched to version 2.2.9 until they fix it

**Comment by daniel-esserweb** [comment]
> @daniel-esserweb Noup, I switched to version 2.2.9 until they fix it

I dont understand why, in the documentation is working since 2.4.0 i think, but in the application hasn't worked yet

**Comment by MarcoRodasW** [comment]
@daniel-esserweb Me neither, I tried different versions and only 2.29 and below worked for me.

**Comment by cnfw** [comment]
I do this temporary workaround to get around the autofocus issue, but it would only work for mouse clicks - keyboard users still need to be considered.

```tsx
<Autocomplete
    onClick={(event) =>
        // @ts-expect-error the input doesn't automatically focus
        event.target.focus()
    }

>
// ...
</Autocomplete>
```

**Comment by kuri-sun** [comment]
@iPagar Thanks for the report! 
Speaking of which, I also found a related bug for Autocomplete component so I will put it down here as well. 
When an user try to remove chars from the input, the input will freeze 😞  (In the video, it's really hard to see but I was pressing the key multiple times.)
These issues seem like happening only on local env(dev env) & sandbox env. 
I will also try to take a look at these! Thanks again for reporting this issue! 👍 

https://github.com/nextui-org/nextui/assets/62743644/04672809-c1e2-4ab6-8bd4-8d950f9872a7

**Comment by cedoromal** [comment]
I think it's the same issue as this one: https://github.com/nextui-org/nextui/issues/3124

It's possible that the focus on input is lost right after the selection is opened. I added a [comment](https://github.com/nextui-org/nextui/issues/3124#issuecomment-2175437744) there for steps to reproduce the focus loss on input.

**Comment by jsonMartin** [comment]
I'm having this same issue. After typing in 1 character, the popup steals focus from the text box, making the user have to click back into the input box to continue typing.

It's a really bad user experience, so until this is fixed I'm rolling back to the last working version where this doesn't happen: `@nextui-org/autocomplete@2.0.10`

**Comment by daniel-esserweb** [comment]
Just a curiosity, it is very buggy in local development but in production it works fine

**Comment by wingkwong** [comment]
@daniel-esserweb it is due to react strict mode, which only happens in local development.

**Comment by MarcoRodasW** [comment]
@daniel-esserweb Does not work in production as well

**Comment by wingkwong** [comment]
@MarcoRodasW are you using 2.4.2? also can you share a minimal reproducible project?

**Comment by ed-quiroga-2103** [comment]
This is almost a year old, but I just ran into this issue and just wanted to leave my findings over here. In my case I was using the `Autocomplete` component inside of an `AccordionItem` component, and the behavior corrected itself when I pulled it out of the accordion. This is kinda specific to my case, but just in case anyone else is containing the `Autocomplete` inside something dynamic, it could be the issue.
